app/views/spree/components/checkout/steps/_payment_step.html.erb

```diff
@@ -1,59 +1,24 @@
-<fieldset>
-  <legend>
-    <%= t('spree.payment_information') %>
-  </legend>
+<div class="payment-step">
+  <fieldset class="payment-step__payment">
+    <legend>
+      <%= t('spree.payment_information') %>
+    </legend>
 
-  <% if @wallet_payment_sources.present? %>
-    <div>
-      <%= radio_button_tag 'use_existing_card', 'yes', true %>
-      <label for="use_existing_card_yes">
-        <%= t('spree.use_existing_cc') %>
-      </label>
-      <%= radio_button_tag 'use_existing_card', 'no' %>
-      <label for="use_existing_card_no">
-        <%= t('spree.use_new_cc_or_payment_method') %>
-      </label>
-    </div>
+    <%= render 'spree/components/checkout/payment_selector', methods: @order.available_payment_methods %>
+    <%= render 'spree/components/checkout/payment_controls', methods: @order.available_payment_methods %>
 
-    <table id="existing_cards">
-      <tbody>
-        <% @wallet_payment_sources.each do |wallet_payment_source| %>
-          <%= render(
-            partial: "spree/checkout/existing_payment/#{wallet_payment_source.payment_source.payment_method.partial_name}",
```

Comment:
I think we should keep this dynamic, each payment method has its own way of handling this, es: https://github.com/solidusio/solidus_stripe/blob/master/lib/views/frontend/spree/checkout/existing_payment/_stripe.html.erb

Reply:
This [commit](https://github.com/nebulab/solidus_starter_frontend/pull/36/commits/d48b11725dc82339d229b20b0e058a3774f85c04) restore the compatibility with solidus_stripe. Hopefully with other payment methods that respect the same insertion points used by solidus_stripe.